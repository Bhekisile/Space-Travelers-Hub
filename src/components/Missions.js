import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  setMissions,
} from '../redux/missions/MissionSlice';
import './styles/Missions.css';

function Missions() {
  const dispatch = useDispatch();
  const missionItem = useSelector((state) => state.missions);
  const joinedMissions = useSelector((state) => state.missions.joinedMissions);

  useEffect(() => {
    const fetchMissions = async () => {
      try {
        const response = await fetch('https://api.spacexdata.com/v3/missions');
        const data = await response.json();
        dispatch(setMissions(data));
      } catch (error) {
        console.error('Error fetching missions:', error);
      }
    };

    fetchMissions();
  }, [dispatch]);
  return (
    { missionItem }
  );
}
export default Missions;
