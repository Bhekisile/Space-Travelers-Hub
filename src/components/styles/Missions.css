.container {
  width: 86%;
  margin: 20px auto;
}

th,
td,
table {
  border: 1px solid rgba(111, 111, 111, 0.6);
  border-collapse: collapse;
}

th {
  padding: 5px 5px;
  text-align: left;
}

td,
table {
  padding: 5px 10px 20px 5px;
}

table tbody tr:nth-child(odd) {
  background-color: #dfdfdf;
}

.mission-name {
  vertical-align: top;
}

.join-button {
  box-sizing: border-box;
  height: 35px;
  font-size: 15px;
  width: 110px;
  border-radius: 5px;
  margin-left: 12px;
}

.mission-description {
  font-size: 16px;
}
