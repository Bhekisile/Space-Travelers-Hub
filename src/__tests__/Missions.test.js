import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Missions from '../components/Missions';

const mockStore = configureStore([]);

describe('Missions component', () => {
  test('renders Missions component', () => {
    const store = mockStore({ missions: { missions: [] } });
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('handle join and leave the mission', () => {
    const store = mockStore({ missions: { missions: [] } });
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
