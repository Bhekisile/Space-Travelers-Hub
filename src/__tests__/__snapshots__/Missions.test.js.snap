// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component handle join and leave the mission 1`] = `
<div>
  <div
    class="container"
  >
    <p
      class="loader"
    >
      Loading...
    </p>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th
            style="width: 130px;"
          >
            Mission
          </th>
          <th>
            Description
          </th>
          <th
            rowspan="3"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`Missions component renders Missions component 1`] = `
<div>
  <div
    class="container"
  >
    <p
      class="loader"
    >
      Loading...
    </p>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th
            style="width: 130px;"
          >
            Mission
          </th>
          <th>
            Description
          </th>
          <th
            rowspan="3"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
